File: src/main/java/org/dedda/bratwurst/lang/ReadVariable.java
package org.dedda.bratwurst.lang;

import org.dedda.bratwurst.lang.scope.Scope;

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
public class ReadVariable extends BWExpression {

    private final String variableName;
    private BWObject value = new BWInteger(0);

    public ReadVariable(int lineNumber, String variableName) {
        super(lineNumber);
        this.variableName = variableName;
    }

    @Override
    public BWObject getValue() {
        return value;
    }

    @Override
    public int getIntValue() {
        return value.getIntValue();
    }

    @Override
    public String getValueType() {
        return value.getValueType();
    }

    @Override
    public void run(Scope scope) {
        BWVariable variable = scope.getVariable(variableName);
        if (variable != null) {
            value = variable.getValue();
        } else {
            throw new RuntimeException("Variable " + variableName + " is not defined!");
        }
    }
}

File: src/main/java/org/dedda/bratwurst/lang/ReadVariable.kt
package org.dedda.bratwurst.lang

import org.dedda.bratwurst.lang.scope.Scope

/**
 * Created by dedda on 10/16/15.
 *
 * @author dedda
 */
class ReadVariable(lineNumber: Int, private val variableName: String) : BWExpression(lineNumber) {
    private var value: BWObject = BWInteger(0)
    override fun getValue(): BWObject {
        return value
    }

    override fun getIntValue(): Int {
        return value.intValue
    }

    override fun getValueType(): String {
        return value.valueType
    }

    override fun run(scope: Scope) {
        val variable = scope.getVariable(variableName)
        value = if (variable != null) {
            variable.value
        } else {
            throw RuntimeException("Variable $variableName is not defined!")
        }
    }

}